observability/microservices.md

```diff
@@ -0,0 +1,53 @@
+# Observability in Microservices
+
+Microservices is a very popular software architecture, where the application is arranged as a collection of loosely coupled services. Some of those services can be written in different languages by different teams.
+
+## Motivations
+
+We need to consider special cases when creating a microservice architecture from the perspective of observability. We want to capture the interactions when making requests between those microservices and correlate them.
+
+Imagine we have a microservice that accesses a database to retrieve some data as part of a request. This microservice is going to be called by someone else as part of an incoming http request or an internal process being executed. What happens if a problem occurs during the retrieval of the data (or the update of the data)? How can we associate, or correlate, that this particular call failed in the destination microservice?
+
+This is a common issue. When calling other microservices, depending on the technology stack we use, we can accidentally hide errors and exceptions that might happen on the other side. If we are using a simple REST interface, the other microservice can return a 500 HTTP status code and we don't have any idea what happen inside that microservice.
+
+More important, we don't have any way to associate our Correlation Id to whatever happens inside that microservice. Therefore, is so important to have a plan in place to be able to extend your traceability and monitoring efforts, especially when using a microservice architecture.
```

Comment:
Is it possible to mention in this section about the way(s) to solve the problems you have mentioned above?

Reply:
One of the things that I mention down there is that most of the framework already support W3C correlation and it's automatically solved.

Reply:
I think what I meant was that there was no transition between the statement of the problem (or motivation to what to solve) and how to solve it, instead it jumped to the implementation on how it works. It is more around the transition, than the content.

Reply:
Technically the next part is about how to solve that problem and understanding at the same time. I mean, we don't want to have technical implementations here, just explain what and why. What would be your ideal mention here? Talking about how to solve it? libraries?